### DISCUSSION TITLE
Future of ControllerX / Appdaemon 15 new config path

### DISCUSSION THREAD

**Discussion by schneekluth** [xaviml/controllerx#874] [General]
Hello @xaviml ,

In the future it will no longer be possible to distribute new versions of ControllerX via HACS. See:
https://github.com/hassio-addons/addon-appdaemon/issues/287

As I am a happy user of ControllerX I'd like to ask if you have a suggestion or an idea on how to distribute future releases?

**Comment by overas** [comment]
Yes, you can distribute new versions of ControllerX via HACS if you edit appdaemon.yaml where appdaemon is going to find the ControllerX app folder.
![image](https://github.com/xaviml/controllerx/assets/37910044/0e2bedfc-2413-444b-a210-ba2cb64ea528)

Read more here:
https://github.com/xaviml/controllerx/issues/875#issuecomment-1921085798

**Reply by schneekluth** [reply]
I am aware of this solution but I am unsure if this is future proof. According to this comment long term goal is to restrict access to add-on locations:

https://github.com/hassio-addons/addon-appdaemon/issues/287#issuecomment-1813112784

**Comment by TheTimeWalker** [comment]
I'm also pretty much struggling to find a good solution to this. From reading multiple threads, issues, etc. it just boils down to:

- You can currently use a workaround by moving the folder, but the support for this will disappear at an undefined point of time ([[link from]()](https://github.com/hassio-addons/addon-appdaemon/issues/287#issuecomment-1813112784))
- Using HACS to install Appdaemon apps means that the Appdaemon addon is "not the best option for you" (https://github.com/hassio-addons/addon-appdaemon/issues/295#issuecomment-1813040519) and that compatibility won't be considered at all (https://github.com/hassio-addons/addon-appdaemon/issues/287#issuecomment-1813110666)
- HACS runs in Home Assistant, so it can't see the addon folders. It has to be an addon itself for this to work (https://github.com/hacs/integration/issues/3408#issuecomment-1868372752)

It sounds like in the future, the easy way of just using HACS to install Appdaemon apps will be deprecated at some point since there does not seem to be any incentive to make them work with each other. And if you're using Appdaemon externally and not as an addon, then making HACS work with that would be a hassle as well. Looks like you would have to resort to using the manual installation guide from any app, like ControllerX, to install it to Appdaemon and to keep up with updates.

**Reply by xaviml** [reply]
Thank you @TheTimeWalker for your analysis, I really appreciate it. I will update documentation to indicate steps on how to install it with HACS (with app_dir parameter) assuming that AppDaemon addon will not restrict the access to the folder in the near future.

The alternative will need to be to create a custom AppDaemon addon, which should not be too complicated to create.

**Reply by xaviml** [reply]
New installation steps have been added to the preview documentation: https://controllerx.netlify.app/controllerx/start/installation/

Hope the steps are more clear and solve the issue that people might have when updating or installing ControllerX for now.
